test/test_eigensolver.cpp

```diff
@@ -0,0 +1,141 @@
+																			 "[10,10]((0, 0, 0, 0, 0, 0, 0, 0, 0, 0),(0, 5.34552e-05, 0, 0, 0, 0, 0, 0, 0, 0),(0, 0, -5.34552e-05, 0, 0, 0, 0, 0, 0, 0),(0, 0, 0, 0, 0, 0, 0, 0, 0, 0),(0, 0, 0, 0, 2.58278e-05, 0, 0, 0, 0, 0),(0, 0, 0, 0, 0, -2.58278e-05, 0, 0, 0, 0),(0, 0, 0, 0, 0, 0, 1.58579e-05, 0, 0, 0),(0, 0, 0, 0, 0, 0, 0, -1.58579e-05, 0, 0),(0, 0, 0, 0, 0, 0, 0, 0, 5.77581e-06, 0),(0, 0, 0, 0, 0, 0, 0, 0, 0, -5.77581e-06))\0",
+																			 "[10,10]((0, 0, 0, 0, 0, 0, 0, 0, 0, 0),(0, 11.8224, 0, 0, 0, 0, 0, 0, 0, 0),(0, 0, -11.8224, 0, 0, 0, 0, 0, 0, 0),(0, 0, 0, 0, 0, 0, 0, 0, 0, 0),(0, 0, 0, 0, 3.47865, 0, 0, 0, 0, 0),(0, 0, 0, 0, 0, -3.47865, 0, 0, 0, 0),(0, 0, 0, 0, 0, 0, 0.5707, 0, 0, 0),(0, 0, 0, 0, 0, 0, 0, -0.5707, 0, 0),(0, 0, 0, 0, 0, 0, 0, 0, 0.0717088, 0),(0, 0, 0, 0, 0, 0, 0, 0, 0, -0.0717088))\0",
+																			 "[10,10]((0, 0, 0, 0, 0, 0, 0, 0, 0, 0),(0, 5.34552e+10, 0, 0, 0, 0, 0, 0, 0, 0),(0, 0, -5.34552e+10, 0, 0, 0, 0, 0, 0, 0),(0, 0, 0, 0, 0, 0, 0, 0, 0, 0),(0, 0, 0, 0, 2.58278e+10, 0, 0, 0, 0, 0),(0, 0, 0, 0, 0, -2.58278e+10, 0, 0, 0, 0),(0, 0, 0, 0, 0, 0, 1.58579e+10, 0, 0, 0),(0, 0, 0, 0, 0, 0, 0, -1.58579e+10, 0, 0),(0, 0, 0, 0, 0, 0, 0, 0, 5.77581e+09, 0),(0, 0, 0, 0, 0, 0, 0, 0, 0, -5.77581e+09))\0"
+																			};
+
+
+int main() {
+
+	typedef double TYPE;
+	typedef float TYPE2;
+
+	typedef matrix<TYPE> MATRIX;
+
+	MATRIX A;
+	MATRIX EVALR;
+	MATRIX EVALC;
+	MATRIX EVECR;
+	MATRIX EVECC;
+	MATRIX GT_EVALR;
+	MATRIX GT_EVALC;
+	MATRIX Zero_Matrix = zero_matrix<TYPE>(3,3);
+
+	int numTestCases = 8;
+
+
+	for (int k = 0; k < numTestCases; k++){
+
+		std::cout << "Running test case# " << (k + 1) << "\n";
+
+		{
+			std::istringstream is(matrix_IN[k]);
+			is >> A;
+		}
+
+		{
+			std::istringstream is(matrix_EVALR[k]);
+			is >> GT_EVALR;
+		}
+
+		{
+			std::istringstream is(matrix_EVALC[k]);
+			is >> GT_EVALC;
+		}
+
+
+		eigen_solver<MATRIX> es(A, EIGVEC);
+		matrix<double> evals_r = es.get_real_eigenvalues();
+		matrix<double> evals_c = es.get_complex_eigenvalues();
+		matrix<double> evecs_r = es.get_real_eigenvectors();
+		matrix<double> evecs_c = es.get_complex_eigenvectors();
+
+
+		assertTrue("Real portion of Eigen Values Match:", compare_on_tolerance(evals_r, GT_EVALR));
+		assertTrue("Imag. portion of Eigen Values Match:", compare_on_tolerance(evals_c, GT_EVALC));
+
+		matrix<std::complex<TYPE> > V(3, 3);
+		matrix<std::complex<TYPE> > D(3, 3);
+		matrix<std::complex<TYPE> > M(3, 3);
+		matrix<std::complex<TYPE> > Lambda;
+
+		MATRIX Lambda_Real(3, 3);
+		MATRIX Lambda_Complex(3, 3);
+
+		for (int i = 0; i < 3; i++){
+			for (int j = 0; j < 3; j++){
+				V(i, j) = std::complex<TYPE>(evecs_r(i, j), evecs_c(i, j));
+				D(i, j) = std::complex<TYPE>(evals_r(i, j), evals_c(i, j));
+				M(i, j) = std::complex<TYPE>(A(i, j), 0.0);
+			} 
+		}
+		Lambda = prod(M, V) - prod(V, D);
```

Comment:
The name `Lambda` here is misleading, because in this context, the letter `lambda` is used to denote a generic eigenvalue of a matrix. Maybe use `Verifier`, since that is what the message below calls it.